crates/protocol/protocol/src/output_root.rs

```diff
@@ -0,0 +1,90 @@
+//! The [`OutputRoot`] type.
+
+use alloy_primitives::{B256, keccak256};
+use derive_more::Display;
+
+/// The [`OutputRoot`] is a high-level commitment to an L2 block. It lifts the state root from the
+/// block header as well as the storage root of the [Predeploys::L2_TO_L1_MESSAGE_PASSER] account
+/// into the top-level commitment construction.
+///
+/// <https://specs.optimism.io/protocol/proposals.html#l2-output-commitment-construction>
+///
+/// [Predeploys::L2_TO_L1_MESSAGE_PASSER]: crate::Predeploys::L2_TO_L1_MESSAGE_PASSER
+#[derive(Debug, Display, Clone, Copy, PartialEq, Eq, Hash)]
+#[display("OutputRootV0({}, {}, {})", state_root, wd_storage_root, block_hash)]
+pub struct OutputRoot {
+    /// The state root of the block corresponding to the output root.
+    pub state_root: B256,
+    /// The storage root of the `L2ToL1MessagePasser` predeploy at the block corresponding to the
+    /// output root.
+    pub wd_storage_root: B256,
```

Comment:
what does wd mean in this context? ah, withdrawals?...better to use longer more descriptive names